wikidict/utils.py

```diff
@@ -373,13 +375,24 @@ def clean(text: str) -> str:
     # Local links
     text = sub(r"\[\[([^||:\]]+)\]\]", "\\1", text)  # [[a]] -> a
 
-    # Files
-    pattern = "|".join(iter(pattern_file))
-    text = sub(rf"\[\[(?:{pattern}):.+?(?=\]\])\]\]*", "", text)
-
-    # More local links
-    text = sub(r"\[\[({{[^}]+}})\]\]", "\\1", text)  # [[{{a|b}}]] -> {{a|b}}
-    text = sub(r"\[\[[^|]+\|(.+?(?=\]\]))\]\]", "\\1", text)  # [[a|b]] -> b
+    # Namespaces
+    # [[File:...|...]] -> ''
+    # except [[File:...|{{...}}]] that will end on '{{...}}'
+    pattern = "|".join(iter(namespaces[locale]))
+    text = sub(rf"\[\[(?:{pattern}):[^\{{]+?(?=\]\])\]\]*", "", text)
```

Comment:
This is a ugly hack for "Annexe" and "Fichier". We want to keep namespaces using at least one template, like `[[Annexe:blablabla|{{foo:|bar}}]]` but bot simple calls like `[[Fichier:blablabla|foo|bar]]`.

It's fragile. If you can find another way, I'm open :)

Reply:
Maybe I found another way.

If we decided to parse those wikicode, and filter out later.
We really need to keep namespaces that doesn't ends with a filename: we could do that check. If it seems to end with a file extension, then fileter out. Else, keep the wikicode.

Reply:
We need to add more examples to be sure about a strategy.